src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -89050,5 +89050,93 @@ class Derived<T> : Base<T> where T : class
             Assert.False(dGoo.Parameters[0].Type.IsNullableType());
             Assert.True(dGoo.Parameters[0].TypeWithAnnotations.NullableAnnotation == NullableAnnotation.Annotated);
         }
+
+        [Fact, WorkItem(29956, "https://github.com/dotnet/roslyn/issues/29956")]
+        public void ConditionalExpression_InferredResultType()
```

Comment:
>ConditionalExpression_InferredResultType [](start = 20, length = 40)

It doesn't look like this code covers all the code paths introduced by the result type inference code in ```NullableWalker.VisitConditionalAccess```. I think we would want at least all branches be covered by targeted tests.

Reply:
See also scenario in [https://github.com/dotnet/roslyn/issues/35075](https://github.com/dotnet/roslyn/issues/35075), even though there is a bug in the binder layer, I think we would want to end up with the "same" result type. Please incorporate it into one of the tests.

---
In reply to: [276453260](https://github.com/dotnet/roslyn/pull/34973#discussion_r276453260) [](ancestors = 276453260)